ethyca/fides
# Escape datetime and ObjectId values in test privacy results

Closes [LA-197](https://ethyca.atlassian.net/browse/LA-197)

### Description Of Changes

Fixes and issue where `ObjectId` values from MongoDB weren't being handled correctly by the `get_test_privacy_request_results` endpoint

### Code Changes

* Escaping raw access results before returning them in the API response

### Steps to Confirm

1. Start Fides `nox -s dev -- mongodb`
2. Create this dataset [mongo_example_test_dataset.txt](https://github.com/user-attachments/files/18031867/mongo_example_test_dataset.txt)
3. Create a system named "MongoDB" and configure the MongoDB integration. Attach the dataset from step 2
```
[mongodb_example]
server="mongodb_example"
user="mongo_user"
password="mongo_pass"
db="mongo_test"
port=27017
```
4. Navigate to http://localhost:3000/systems/configure/mongodb/test-datasets
5. Enter the following as the `Test inputs`
```
{
  "email": "customer-2@example.com",
  "postgres_example_test_dataset:customer:id": 1
}
```
6. Hit run and verify the request is successful

### Pre-Merge Checklist

* Followup issues:
* Database migrations:
* Documentation:


[LA-197]: https://ethyca.atlassian.net/browse/LA-197?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
